flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-paimon/src/main/java/org/apache/flink/cdc/connectors/paimon/sink/v2/PaimonWriter.java

```diff
@@ -92,21 +96,29 @@ public PaimonWriter(
     }
 
     @Override
-    public Collection<MultiTableCommittable> prepareCommit() throws IOException {
+    public Collection<MultiTableCommittable> prepareCommit() {
         List<MultiTableCommittable> committables = new ArrayList<>();
-        for (Map.Entry<Identifier, StoreSinkWrite> entry : writes.entrySet()) {
-            Identifier key = entry.getKey();
-            StoreSinkWrite write = entry.getValue();
-            boolean waitCompaction = true;
-            committables.addAll(
-                    // here we set it to lastCheckpointId+1 to
-                    // avoid prepareCommit the same checkpointId with the first round.
-                    write.prepareCommit(waitCompaction, lastCheckpointId + 1).stream()
-                            .map(
-                                    committable ->
-                                            MultiTableCommittable.fromCommittable(key, committable))
-                            .collect(Collectors.toList()));
-        }
+        writes.entrySet().parallelStream()
+                .forEach(
```

Comment:
Since ArrayList is not thread-safe, maybe we can use flapMap().collect(Collectors.toList()) instead of forEach to reduce thread safety issues.

Reply:
Done.